File: Week 1 - 03.02.2021/Selection sort 19.py
# Selection Sort using Python
# Author : Abhishek Sharma

def selection_sort(user, s, n):
    for i in range (0,s):
        min_val = user[i]
        addr = i
        for j in range (i+1,n):
            if user[j]<min_val:
                min_val = user[j]
                addr = j
        temp = user[i]
        user[i]=user[addr]
        user[addr]=temp
        

first = list(map(int, input().split(" ")))
n = first[0]
s = first[1]
user = list(map(int, input().split(" ")))
selection_sort(user,s,n)
for k in range (0,n):
    print (user[k], end = " ")

File: Week 1 - 03.02.2021/Selection Sort 19.cpp
// Selection Sort using C++
// Author : Abhishek Sharma

#include<iostream>
 
using std::cout;
using std::cin;
using std::endl;
 
void insertionsort(int p[],int no_of_steps,int n)
{
    int min_ele;
    int addr;
    int temp;
 
    for (int i = 0; i<no_of_steps; ++i)
    {
 
        min_ele = p[i];
        addr = i;
 
        for(int j=i+1;j<n;++j)
        {
            if (p[j]<min_ele)
            {
                min_ele = p[j];
                addr = j;
            }
        }
 
        temp = p[i];
        p[i] = p[addr];
        p[addr] = temp;
    }
 
}
 
int main()
{
    int n,no_of_steps;
    cin>>n>>no_of_steps;
    int a[n];
    for(int i=0;i<n;++i)
    {
        cin>>a[i];
    }
 
    // cout<<"output:"<<endl<<endl;
 
    // for(int i=0;i<n;++i)
    // {
    //     cout<<a[i]<<' ';
    // }
 
    insertionsort(a,no_of_steps,n);
 
    for(int i=0;i<n;++i)
    {
        cout<<a[i]<<' ';
    }
 
    return 0;
}
